pkg/resources/reconciler.go

```diff
@@ -130,6 +130,18 @@ func (r *Reconciler) ReconcileNamespace(ctx context.Context, namespace string, i
 		return integreatlyv1alpha1.PhaseCompleted, nil
 	}
 
+	if inst.Spec.PullSecret.Name != "" {
+		_, err := r.ReconcilePullSecret(ctx, namespace, inst.Spec.PullSecret.Name, inst, client)
```

Comment:
IIRC, there are some products are calling this function atm.  If we are reconciling the pull secret here, and this function is called for all products, then I think we can remove the existing references to this function.

Reply:
We could, but when this function is called with the product reconcilers it's called with a reference to a specific secret like here https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/apicurito/reconciler.go#L130 whereas how I've used it we are checking for a secret in the installation spec and using that.

I have no objection to removing the function from the other products as you've said it doesn't seem to be used for anything atm but as I'm doing something different here than how it's currently used, I left in what was there.

Reply:
I agree, I'm not sure would it be alright to remove this call in the individual product reconcilers as originally it was added as part of https://issues.redhat.com/browse/INTLY-3511 to fix starting walkthrough 1 correctly as they might expect pull secrets with a specifc name in the namespace or would it be fine with the service accounts linked to the pull secrets now :thinking:
@JameelB might be able to better comment on this

Would be good to make sure the walkthroughs that use these products still work correctly if we do remove these references.

However, if we don't remove these references and the `PullSecret` is not intially set in the RHMI CR, I believe it will be set to the default pull secret from any one of the products that call `ReconcilePullSecret()`. Then this part of the code will execute on next namespace reconcile, copying and linking this pull secret to the namespace and service accounts. Just wondering is this what you are expecting? :thinking:  
https://github.com/integr8ly/integreatly-operator/blob/6fecb88f02d8bc16d33270a73f8be26ad511ab65/pkg/resources/pullSecret.go#L22-L31

Reply:
@KevFan is correct. As far as I remember, Fuse was the only one that expected the secret to be there. It has to be named a certain way as well. If that secret is missing, I think the fuse operator reconcile will fail if it can't find that secret even if the pull secret is already attached to the service account. There is a new version of Fuse coming up. I'm not exactly sure if this is still the case when we upgrade.

I don't think we have this issue for the other products though so we can probably remove it there. As @KevFan best to check the walkthroughs and make sure that they work.

Reply:
@KevFan

> However, if we don't remove these references and the PullSecret is not intially set in the RHMI CR, I believe it will be set to the default pull secret from any one of the products that call ReconcilePullSecret(). Then this part of the code will execute on next namespace reconcile, copying and linking this pull secret to the namespace and service accounts. Just wondering is this what you are expecting?

From what I can see, it does set some default values for the fields in the CR, however, they are not saved as part of this function, so I don't know if they will have any impact on the next reconciliation. Also I don't know why we set these default values: just coping the default secrets to a namespace does nothing. I feel like we should just review the logic to set the default values.

Reply:
> From what I can see, it does set some default values for the fields in the CR, however, they are not saved as part of this function, so I don't know if they will have any impact on the next reconciliation. Also I don't know why we set these default values: just coping the default secrets to a namespace does nothing. I feel like we should just review the logic to set the default values.

@wei-lee Yeah, it's not saved as part of this function, but I'm pretty sure it does get saved into the CR further down the call chain somewhere. Maybe by the controller here:
https://github.com/integr8ly/integreatly-operator/blob/03ae73c39d3b917bf2bf10f0c8cd54f97eb8410f/pkg/controller/installation/installation_controller.go#L353
Also had a quick run of this PR yesterday, and it looks like the case. But I do agree on reviewing this logic for setting the default values